blockchain/chain/src/store/chain_store.rs

```diff
@@ -106,4 +107,53 @@ impl<'a> ChainStore<'a> {
     pub fn blockstore(&self) -> &Blockstore {
         &self.db
     }
+    /// Returns Tipset from key-value store from provided keys
+    pub fn tipset(&self, keys: &[Cid]) -> Result<Tipset, Error> {
```

Comment:
Optional, to maybe have `keys` be `cids` since it makes more sense than referring to it as the key value for the db. Also kinda confusing below to call k.key()